idaes/power_generation/carbon_capture/mea_solvent_system/property_package/liquid_prop.py

```diff
@@ -0,0 +1,1282 @@
+
+        self.mw_ave = Expression(rule = rule_mw_ave,
+                            doc ="Average molecular weight of  liquid phase"
+                            "components [kg/mol]" )
+
+    def _mass_frac(self):
+        def rule_mass_frac(blk,j):
+            return  blk.mole_frac[j]*blk._params.mw[j]/blk.mw_ave
+
+        self.mass_frac = Expression(self._params.component_list,
+                            rule = rule_mass_frac,
+                            doc ="weight fraction"
+                            "[-]" )
+
+    def _mass_frac1(self):
+        def rule_mass_frac1(blk,j):
+            return  blk.mole_frac[j]*blk._params.mw[j]/\
+                    sum(blk.mole_frac[i]*blk._params.mw[i]
+                       for i in blk._params.component_list_solvent)
+
+        self.mass_frac1 = Expression(self._params.component_list_solvent,
+                            rule = rule_mass_frac1,
+                            doc ="weight fraction on CO2 free basis"
+                            "[-]" )
+
+    def _vol_mol_comp(self):
+        def rule_vol_mol_comp(blk,j):
+            T  = blk.temperature
+            mw = blk._params.mw
+            c  = blk._params.vol_mol_comp_param
+            return mw[j]*1e3/(c[j,1]*T**2 +c[j,2]*T + c[j,3])
+
+        self.vol_mol_comp = Expression(self._params.component_list_solvent,
+                                      rule = rule_vol_mol_comp,
+                                 doc="Pure solvent molar volume [cm^3/mol]")
+
+    def _vol_mol(self):
+        # Liquid phase molar volume m^3/mol
+        def rule_vol_mol(blk):
+            v_mea = self.vol_mol_comp['MEA']
+            v_h20 = self.vol_mol_comp['H2O']
+            a = blk._params.vol_mol_param[1]
+            b = blk._params.vol_mol_param[2]
+            c = blk._params.vol_mol_param[3]
+            d = blk._params.vol_mol_param[4]
+            e = blk._params.vol_mol_param[5]
+            v_soln = (blk.mole_frac['MEA']*v_mea + blk.mole_frac['H2O']*v_h20 +
+                      a * blk.mole_frac['CO2'] + (b + c * blk.mole_frac['MEA'])
+                      * blk.mole_frac['MEA']*blk.mole_frac['H2O'] +
+                      (d + e * blk.mole_frac['MEA']) *
+                      blk.mole_frac['MEA']*blk.mole_frac['CO2'])*1e-6
+            return v_soln
+
+        self.vol_mol = Expression(rule = rule_vol_mol,
+                                 doc="liquid molar volume [m3/mol]")
+
+    def _conc(self):
+        # Liquid phase concentration
+        def rule_conc(b):
+            return 1/self.vol_mol
+
+        self.conc = Expression(rule = rule_conc, doc="concentration [mol/m3]" )
+
+    def _conc_comp(self):
+        def rule_conc_comp(blk,i):
+            return blk.conc*blk.mole_frac[i]
+
+        self.conc_comp = Expression(self._params.component_list,
+                        rule = rule_conc_comp, doc="concentration of "
+                        "liquid components [mol/m3]" )
+
+    def _lnk_eq_conc(self):
```

Comment:
What triggers this method? It appears to be a method attached to the State Block, which means that something needs to call it before this will be constructed. If this is called by something else in the State Block, then a comment would be good here and I would consider moving this to be part of that method for clarity.

Reply:
Yes, it's a method attached to the state Block where anything that needs it can call it before it is constructed. In addition, calling the property directly(self.lnk_eq_conc) within the state block will trigger the method(expect in the future you are planning to stop this behavior).

Reply:
My main question was where was this method being called from, as it is not clear from the method alone. A comment explaining places where this method is called would be helpful.